picamera2/picamera2.py

```diff
@@ -1119,10 +1119,11 @@ def process_requests(self) -> None:
             display_request = self.completed_requests[-1]
             display_request.acquire()
 
-            # Some applications may (for example) want us to draw something onto these images before
-            # encoding or copying them for an application.
-            if display_request and self.pre_callback:
-                self.pre_callback(display_request)
+            for req in requests:
+                # Some applications may (for example) want us to draw something onto these images before
+                # encoding or copying them for an application.
+                if self.pre_callback:
```

Comment:
Very tiny nit-pick... maybe do this check outside the loop?

Reply:
Good point, definitely makes sense to do it the other way round.  (Will be back in on Wednesday and will sort then)